packages/app-mobile/components/NoteEditor/CodeMirror/RegionSpec.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * Describes an inline region's formatting. For example, an italicized region.
+ */
+
+import { Text as DocumentText, SelectionRange } from '@codemirror/state';
+
+export default class RegionSpec {
+	// Patterns to use when creating an instance of the tag.
+	// E.g.
+	//   templateStart=**
+	//   templateStop=**
+	//  would be used to create
+	//   **content here**
+	public templateStart: string;
+	public templateStop: string;
+
+	// Size of the buffer used to check for a match.
+	// Particularly useful for [startExp], [stopExp].
+	// Only used when the selection is empty.
+	protected startBuffSize: number;
+	protected stopBuffSize: number;
+
+	// Regular expressions for matching possible starting/stopping
+	// regions.
+	protected startExp?: RegExp;
+	protected stopExp?: RegExp;
+
+	public constructor({
+		templateStart, templateStop,
+		startExp, stopExp,
+		startBuffSize, stopBuffSize,
+	}: {
+			templateStart: string; templateStop: string;
+			startExp?: RegExp; stopExp?: RegExp;
+			startBuffSize?: number; stopBuffSize?: number;
+	}) {
```

Comment:
Is that what CodeMirror expects? It's so clunky. Just a list of arguments would be enough without that weird double-definition.